src/report-embedded/report-embedded-view.controller.js

```diff
@@ -0,0 +1,59 @@
+/*
+ * This program is part of the OpenLMIS logistics management information system platform software.
+ * Copyright © 2017 VillageReach
+ *
+ * This program is free software: you can redistribute it and/or modify it under the terms
+ * of the GNU Affero General Public License as published by the Free Software Foundation, either
+ * version 3 of the License, or (at your option) any later version.
+ *  
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. 
+ * See the GNU Affero General Public License for more details. You should have received a copy of
+ * the GNU Affero General Public License along with this program. If not, see
+ * http://www.gnu.org/licenses.  For additional information contact info@OpenLMIS.org. 
+ */
+
+(function() {
+
+    'use strict';
+
+    /**
+     * @ngdoc controller
+     * @name report.controller:SupersetReportController
+     *
+     * @description
+     * Controller for superset report view.
+     */
+    angular
+        .module('report')
+        .controller('reportEmbeddedViewController', ReportEmbeddedViewController);
+
+    ReportEmbeddedViewController.inject = ['$scope', '$stateParams', 'loadingModalService', 'reportEmbeddedService'];
+
+    function ReportEmbeddedViewController($scope, $stateParams, loadingModalService, reportEmbeddedService,
+                                          $sce) {
+        var vm = this;
+
+        vm.reportUrl = undefined;
+        vm.isReady = false;
+
+        function onInit() {
+            loadingModalService.open();
+
+            reportEmbeddedService.get($stateParams.id)
+                .then(function(report) {
+                    vm.reportUrl = report.url;
+                    vm.isReady = true;
+
+                    loadingModalService.close();
+                });
+        }
+
+        $scope.trustSrc = function(src) {
+            return $sce.trustAsResourceUrl(src);
+        };
+
+        onInit();
```

Comment:
I think it would be better to use angular's onInit instead of executing it yourself.

`vm.$onInit = ...`